Update CONTRIBUTING.md to describe CHANGELOG convention (#810)
higlass/higlass: CONTRIBUTING.md

- **JavaScript coding style**: We are following a combination of [ESLint's recommendation](https://eslint.org/docs/rules/), [Airbnb's coding style](https://github.com/airbnb/javascript), [React's recommendation](https://github.com/yannickcr/eslint-plugin-react), and [customizations](https://github.com/higlass/higlass/blob/develop/.eslintrc#L30). The easiest way to ensure you're following our coding style is to automatically run `npm run lint`.

- **Python coding style**: We follow the [PEP 8](https://www.python.org/dev/peps/pep-0008/) style guide and the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) convention for [docstrings](https://www.python.org/dev/peps/pep-0257/).

> - **Changelog**: PRs should contain an update to the CHANGELOG.md describing what they change. Exceptions to this convention are acceptable when the changes are strictly developer facing, such as updates to dependencies.

Reviewer: ```suggestion
- **Changelog**: PRs should contain an update to the CHANGELOG.md describing the user-facing changes. Exceptions to this convention are acceptable when the changes are strictly developer facing, such as updates to dependencies.
```

Reviewer 2: A major dependency-version shift (such as with Pixijs, if I remember correctly) could introduce side effects, such that a change might be worth mentioning in the CHANGELOG if it could affect use of higlass components by other groups.
